app/src/test/java/com/losandroides/learn/presentation/MainViewModelTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.losandroides.learn.presentation
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.losandroides.learn.domain.ItemsUseCase
+import com.losandroides.learn.domain.model.Item
+import com.losandroides.learn.framework.coVerifyOnce
+import com.losandroides.learn.framework.relaxedMockk
+import io.mockk.coEvery
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.*
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class MainViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private val testDispatcher = StandardTestDispatcher()
+    private val testCoroutineScope = TestScope(testDispatcher)
+
+    private val itemsUseCase: ItemsUseCase = relaxedMockk()
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @After
+    fun cleanup() {
+        Dispatchers.resetMain()
+    }
```

Comment:
For this is recommended using rules, check my example -> [here](https://github.com/santimattius/android-arch-template/blob/master/app/src/test/java/com/santimattius/template/presentation/viewmodels/HomeViewModelTest.kt#L26)

Reply:
Sure I could wrap it in a rule. But [TestCoroutineDispatcher](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-coroutine-dispatcher/) is deprecated.
And `testDispatcher.cleanupTestCoroutines()` [is not required anymore](https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/MIGRATION.md) when using `StandardTestDispatcher`.

Reply:
No doubt! I'll do it